### ISSUE TITLE
Long description crashes svg export

### ISSUE THREAD

**Issue by Che4ter** [mitre-attack/attack-navigator#224]
When a layer contains a longer description like below and the user tries to export the layer as SVG, the page becomes unresponsive until the browse tab crashes.

`    "description": "arcu cursus vitae congue mauris rhoncus aenean vel elit scelerisque mauris pellentesque pulvinar pellentesque habitant morbi tristique senectus et netus",`

The problem probably lies in the function `optimalFontSize`, where the number of words is used as exponent in the for loop, which leads to a high number of loops.
https://github.com/mitre-attack/attack-navigator/blob/dbafacc4614838073db487e8718b0300aeee8dec/nav-app/src/app/exporter/exporter.component.ts#L265

**Comment by isaisabel** [comment]
You're right that it's that combinatorics in the font size optimizer causing the issue. In the example text you gave, it tries over 500,000 combinations of spaces (each of which requires modification of the DOM). 

In testing, it seems that the threshold where it seems to take a subjectively unreasonable amount of time to render is at about 1000 combinations, giving a max word length of about 11 if I've done my math right (your example has a length of 19).

Skipping some combinations when greater than 1000 combinations would be tried is probably the best "simple" solution:

```js
//optimization: avoid a complete search by only trying a few of the possible combinations
let combinations = 2**(words.length - 1); // how many combinations of spaces there are
let searchGranularity = 1000; //try this number of combinations of spaces
let increment = combinations < searchGranularity? 1 : Math.round(combinations/searchGranularity) // how many combinations to skip
// console.log("combinations:", combinations, 'increment:', increment)
for (let j = 0; j < combinations; j += increment) { //try no breaks first
```

Unfortunately, it doesn't work great in practice, for example here's the description with a word length of 68:

![image](https://user-images.githubusercontent.com/28868888/94019466-d4bc3700-fd7f-11ea-95cc-66a83f65435b.png)

This is obviously nowhere close to an optimal placement of word breaks. We might need to develop a more robust solution than simply skipping combinations that is still more efficient than a complete search. It's possible that some sort of genetic algorithm could breed a solution, but I'm not sure if a genetic algorithm would be able to produce a viable candidate if we can only perform 1000 fitness checks. Might be worth a shot though.

**Comment by isaisabel** [comment]
I ended up fixing this by limiting the number of line breaks it attempts to place.
- Allows 3 breaks for sentences with words < 20
- Allows 2 breaks for sentences with 20 <= words < 50
- Allows 1 break for sentences withwords <= 50
This should work fine roughly up to 1000 words in the description, which is probably never going to happen. 

I've merged the fix into develop. It'll be included in the live instance in the v4.0 release.
